mu4e/mu4e-icalendar.el

```diff
@@ -0,0 +1,150 @@
+                                       mu4e-icalendar-diary-file))))))
+
+(defun mu4e~icalendar-delete-citation ()
+  (delete-region (point-min) (point-max)))
+
+(defun mu4e~icalendar-trash-message (msg)
+  "Trash the message MSG."
+  (with-current-buffer mu4e~headers-buffer-name
+    (let* ((docid (mu4e-message-field msg :docid))
+           (markdescr (assq 'trash mu4e-marks))
+           (action (plist-get (cdr markdescr) :action))
+           (target (mu4e-get-trash-folder msg)))
+      (run-hook-with-args 'mu4e-mark-execute-pre-hook 'trash msg)
+      (funcall action docid msg target))))
+
+(defun mu4e-icalendar-reply-ical (original-msg event status buffer-name)
+  (let ((message-signature nil))
+    (let ((mu4e-compose-cite-function #'mu4e~icalendar-delete-citation)
+          (mu4e-sent-messages-behavior 'delete)
+          (mu4e-compose-reply-recipients 'sender))
+      (mu4e~compose-handler 'reply original-msg))
+    ;; Make sure the recipient is the organizer
+    (let ((organizer (gnus-icalendar-event:organizer event)))
+      (unless (string= organizer "")
+        (message-goto-to)
+        (delete-region (line-beginning-position) (line-end-position))
+        (insert "To: " organizer)))
+    (message-goto-body)
+    (mml-insert-multipart "alternative")
+    (mml-insert-part "text/plain")
+    (let ((reply-event (gnus-icalendar-event-from-buffer
+                        buffer-name mu4e-user-mail-address-list)))
+      (insert (gnus-icalendar-event->gnus-calendar reply-event status)))
+    (forward-line 1); move past closing tag
+    (mml-attach-buffer buffer-name "text/calendar; method=REPLY; charset=utf-8")
+    (message-goto-subject)
+    (delete-region (line-beginning-position) (line-end-position))
+    (insert "Subject: " (capitalize (symbol-name status))
+            ": " (gnus-icalendar-event:summary event))
+    (set-buffer-modified-p nil); not yet modified by user
+    (when mu4e-icalendar-trash-after-reply
+      ;; `mu4e~switch-back-to-mu4e-buffer' was executed.
+      (push
+       (lexical-let ((msg original-msg))
+         #'(lambda ()
+             (when (and (not (eq mu4e-split-view 'single-window))
+                        (buffer-live-p (mu4e-get-view-buffer)))
+               (switch-to-buffer (mu4e-get-view-buffer))
+               (or (mu4e-view-headers-next)
+                   (kill-buffer-and-window))
+               (mu4e~icalendar-trash-message msg))))
+         message-send-actions))
+;    (message-send-and-exit)
+    ))
+
+
+(defun mu4e~icalendar-insert-diary (event reply-status filename)
+  "Insert a diary entry for the EVENT with reply STATUS in FILE."
+  ;; FIXME: handle recurring events
+  (let* ((beg (gnus-icalendar-event:start-time event))
+         (beg-date (format-time-string "%d/%m/%Y" beg))
+         (beg-time (format-time-string "%H:%M" beg))
+         (end (gnus-icalendar-event:end-time event))
+         (end-date (format-time-string "%d/%m/%Y" end))
```

Comment:
Why this specific date-format?

Reply:
For no good reason except that it is the first one [advertised for the diray format](https://www.gnu.org/software/emacs/manual/html_node/emacs/Format-of-Diary-File.html).  Note that it is also different from the date generated by `calendar` when pressing `i d` which makes those entries easily recognizable.

Reply:
Okay, but shouldn't gnus-icalender know what format we're using? Guess I don't fully understand how the date-string is used... Otherwise , I'd prefer ISO dates.